src/stats/stats.jl

```diff
@@ -0,0 +1,158 @@
+export ExecutionStats, statshead, statsline
+
+const STATUS = Dict(:unknown => "unknown",
+                    :first_order => "first-order stationary",
+                    :max_eval => "maximum number of function evaluations",
+                    :max_time => "maximum elapsed time",
+                    :max_iter => "maximum iteration",
+                    :neg_pred => "negative predicted reduction",
```

Comment:
This kind of status will only apply to certain solvers. We should have a mechanism for the solver to set its status if it's not one of the default statuses.

Reply:
I was thinking we should enforce the allowed statuses, but we should expand the list. I just haven't put much thought in other statuses yet.